src/restrictingGuards.ts

```diff
@@ -0,0 +1,97 @@
+import {checkerToGuard} from './Checker';
+import {thenGuard, andGuard, orGuard} from './combinators';
+import {ReasonGuard} from './ReasonGuard';
+import {isNumber, isSymbol, isString} from './primitiveGuards';
+
+export const numberIsInteger = checkerToGuard<number, number>((input: number) => {
+	if (!Number.isInteger(input)) {
+		throw new Error(`${input} is not an integer`);
+	}
+	return `${input} is an integer`;
+});
+
+export const numberIsLessThan = (maximum: number) =>
+	checkerToGuard<number, number>((input: number) => {
+		if (input >= maximum) {
+			throw new Error(`${input} >= ${maximum}`);
+		}
+		return `${input} < ${maximum}`;
+	});
+
+export const numberIsGreaterThan = (minimum: number) =>
+	checkerToGuard<number, number>((input: number) => {
+		if (input <= minimum) {
+			throw new Error(`${input} <= ${minimum}`);
+		}
+		return `${input} > ${minimum}`;
+	});
+
+export const numberIsAtMost = (maximum: number) =>
```

Comment:
Might be good to export copies of these as e.g.  `numberIsLessThanOrEqual`

Reply:
Can do. Wordy as hell :/